File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/CustomGeometrySourceCallback.java
package org.maplibre.nativeffi;

/** Callback invoked by native custom geometry sources for tile fetches and cancels. */
public interface CustomGeometrySourceCallback {
  void fetchTile(CanonicalTileId tileId);

  default void cancelTile(CanonicalTileId tileId) {}
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/style/CustomGeometrySourceCallback.kt
package org.maplibre.nativeffi.style

import org.maplibre.nativeffi.geo.CanonicalTileId

/** Callback invoked by native custom geometry sources for tile fetches and cancels. */
public interface CustomGeometrySourceCallback {
  public fun fetchTile(tileId: CanonicalTileId)

  public fun cancelTile(tileId: CanonicalTileId) {}
}
